pkg/volume/util/operationexecutor/operation_generator.go

```diff
@@ -1462,18 +1476,15 @@ func (og *operationGenerator) GenerateVerifyControllerAttachedVolumeFunc(
 
 }
 
-func (og *operationGenerator) verifyVolumeIsSafeToDetach(
-	volumeToDetach AttachedVolume) error {
+func (og *operationGenerator) verifyVolumeIsSafeToDetachAndUpdateNodeStatus(
+	volumeToDetach AttachedVolume, actualStateOfWorld ActualStateOfWorldAttacherUpdater, needVerify bool) error {
 	// Fetch current node object
-	node, fetchErr := og.kubeClient.CoreV1().Nodes().Get(string(volumeToDetach.NodeName), metav1.GetOptions{})
-	if fetchErr != nil {
-		if errors.IsNotFound(fetchErr) {
-			klog.Warningf(volumeToDetach.GenerateMsgDetailed("Node not found on API server. DetachVolume will skip safe to detach check", ""))
-			return nil
-		}
-
-		// On failure, return error. Caller will log and retry.
-		return volumeToDetach.GenerateErrorDetailed("DetachVolume failed fetching node from API server", fetchErr)
+	var fetchErr error
+	var node *v1.Node
+	if og.nodeLister != nil {
+		node, fetchErr = og.nodeLister.Get(string(volumeToDetach.NodeName))
+	} else {
+		node, fetchErr = og.kubeClient.CoreV1().Nodes().Get(string(volumeToDetach.NodeName), metav1.GetOptions{})
 	}
```

Comment:
I am not sure if we need this if else. I thought the reason this code relied on using `kubeClient` directly vs using informer is because these checks are sensitive to informer caching and reading a cached value can result in incorrect results.

Reply:
+1 let's change one thing at a time. Since this is not necessary for the fix, let's leave it as is.